department-of-veterans-affairs/caseflow
# BillyS/APPEALS-7681: Create Document ready for Board intake review task action

Resolves [APPEALS-7681](https://vajira.max.gov/browse/APPEALS-7681)

### Description
The Create Document ready for Board Intake Review task action modal was added to allow appeal to be sent to board intake for docketing. As a CSP user, I need the option to send appeals to board intake when they are ready to be docketed.

### Acceptance Criteria
- [ ] The following occurs when the VHA CSP user selects the "Documents ready for Board Intake review" task action
   - [ ] A modal appears to confirm intent
   - [ ] Modal Header: Ready for review
   - [ ] Modal information text: This appeal will be sent to Board intake for review. Please select where the documents for this 
          appeal are stored.
   - [ ] User is required to select where the document is stored from the following options
      - [ ]  VBMS
      - [ ]  Centralized Mail Portal
        - [ ] When 'Other' is selected, user is required to indicate the source via additional text entry box as shown in designs
        - [ ] Field type: required free text field  
 - [ ] User may add additional details on file structure or file path
    - [ ] Optional free text field
 - [ ] Modal contains the following selectable options
    - [ ] Cancel
      - [ ] User is returned to case details screen and selection is not retained
    - [ ] Send
      - [ ] Select to execute task action
      - [ ] Option disabled until required field(s) populated
- [ ] The following occurs when the "Documents ready for Board Intake review" task action is executed
   - [ ] The appeal appears on the Completed tab of the VHA CSP predocket queue
   - [ ] The appeal appears on the Ready for Review tab of the BVA Intake organization queue
   - [ ] User is returned to the VHA CSP organization queue
     - [ ] A message appears confirming the case has been sent to Board Intake for review
### Testing Plan
1. Open Caseflow App
2. Switch user to one of the new Caregiver users
![image](https://user-images.githubusercontent.com/98414141/182457704-7b5dbe54-8745-489b-bb46-c5964ad0b2a5.png)
3. Select Your Queue

![image](https://user-images.githubusercontent.com/98414141/182458385-e6e710b3-f3b8-41ed-b8f6-e090de6e481d.png)

4. Switch to the CSP organization's queue:
<img width="852" alt="switch-to-csp-queue" src="https://user-images.githubusercontent.com/99351305/181598516-51ff9792-ba56-4bcd-b9fd-e8aa3776954e.PNG">

5. You will end up in the Caregiver Queue with the completed tab and in-progress tabs being visible so far:
 
![TwoTabs](https://user-images.githubusercontent.com/98414141/184995399-a3693cf1-d8a9-4e97-834e-572da3cf8d31.PNG)

```ruby
6. Open the Rails console (`make c` in VSCode Terminal)

7. Create appeal with a Pre-Docket Task
     appeal = FactoryBot.create(:appeal, :with_pre_docket_task)

8. Check task tree to make sure it created a root task and Pre-docket Task
    appeal.treee

    should display:
```
   ![image](https://user-images.githubusercontent.com/98414141/182461698-60c4200d-7d46-4037-bf40-80c126193936.png)
```ruby 
    NOTE the ID on the PreDocketTask for future use.  The example displayed is 7134
 
9. Define the task by the Task ID
    task=Task.find(7134)

10. Create the Vha Document Search Task
      vha_document_search_task=VhaDocumentSearchTask.create!( parent: task, appeal: appeal, assigned_at: Time.zone.now, assigned_to: VhaCaregiverSupport.singleton )

11. Check task tree to make sure it created a Vha Docuement Search Task
      appeal.treee

     should display:
```
   ![image](https://user-images.githubusercontent.com/98414141/182461939-10a4fdea-c6d9-4882-9a01-f0e23b9ef062.png)

      This time, please NOTE the ID on the VhaDocketSearchTask for future use.  The example displayed is 
       7135. In addition, note that the status on the VhaDocumentSearchTask is "assigned"
```ruby
12.  I find this next step of setting the veteran_file_number helpful for easy identification.  
       appeal.update( veteran_file_number: "500000101" )

13. Go back to Caseflow and refresh the page. Since this appeal is in "assigned" status, 
      It will not appear on either the "in_progress" or "completed" tabs that are showing. 
      Notice the absence of the appeal we created: 
```
![In_progress](https://user-images.githubusercontent.com/98414141/184996839-fac7c469-5f15-46cd-a800-0da9aec56b44.PNG)
![Completed](https://user-images.githubusercontent.com/98414141/184996860-953bc3b7-ea5d-4d03-b748-937a403cd3a9.PNG)
```ruby
14. For testing, there are two ways to handle this:       
       A. You can bypass the "assigned" tab and go directly to the case details page for that appeal 
           with the UUID:
           appeal.uuid
           => "bf4e97d4-4909-4749-a864-d83997d0ff97" # This is just an example
       Case Details page: `/queue/appeals/<appeal.uuid>`
       
       B. Alternatively, you can change the status of the task in the rails console to "in_progress" 
           Task.find(7135).in_progress!
           Go back to Caseflow and refresh the page.  
           Now, this appeal will be visible in the "in progress" tab 
```
![in_prog_appeal](https://user-images.githubusercontent.com/98414141/184998532-736c6b3a-1710-4701-9965-a47de55c88c0.PNG)
```ruby
       Click on the appeal that was created
       Either method will take you to the same place on the case details page

```
![Case Details (3)](https://user-images.githubusercontent.com/98414141/184991827-e6335497-85ce-4951-8264-0296140efe0f.PNG)

15. Choose the one Task Action Available, "Documents ready for Board Intake Review" to open the modal.  Please note the modal screenshots including the button validation and formatting

![Disabled Modal (4)](https://user-images.githubusercontent.com/98414141/184992790-9b868bd2-f67c-4759-bbe3-32dc33f1aa72.PNG)
![Enabled Modal (3)](https://user-images.githubusercontent.com/98414141/184992361-bc11b0b5-0b39-4ce6-8fe9-b381fda9b7cf.PNG)
![Other Disabled Modal (3)](https://user-images.githubusercontent.com/98414141/184992392-1c388d5b-ef4e-4d4b-9b66-49edcac8f6ad.PNG)
![Other Enabled Modal (3)](https://user-images.githubusercontent.com/98414141/184992406-c71dbd3b-1210-4d2b-992b-edfffd9402e2.PNG)

16. Choose one of the radio buttons and hit send
17. This confirmation message will be displayed
![Confirmation](https://user-images.githubusercontent.com/98414141/184993320-9995182b-4aeb-43c6-85e3-d860469296d2.PNG)

18. Click on the Completed tab.  Now the case should be visible   
![Completed Tab](https://user-images.githubusercontent.com/98414141/184993802-0c3f88bf-effc-479a-b746-40228cbd3557.PNG)
